synapse/rest/admin/rooms.py

```diff
@@ -155,3 +160,68 @@ def __init__(self, hs):
                 "new_room_id": new_room_id,
             },
         )
+
+
+class ListRoomRestServlet(RestServlet):
+    """
+    List all rooms that are known to the homeserver. Results are returned
+    in a dictionary containing room information. Supports pagination.
+    """
+
+    PATTERNS = admin_patterns("/rooms")
+
+    def __init__(self, hs):
+        self.store = hs.get_datastore()
+        self.auth = hs.get_auth()
+        self.admin_handler = hs.get_handlers().admin_handler
+
+    async def on_GET(self, request):
+        requester = await self.auth.get_user_by_req(request)
+        await assert_user_is_admin(self.auth, requester.user)
+
+        # Extract query parameters
+        start = parse_integer(request, "from", default=0)
+        limit = parse_integer(request, "limit", default=100)
+        order_by = parse_string(request, "order_by", default="alphabetical")
+        if RoomSortOrder(order_by) not in (
```

Comment:
Will this not raise an exception if it isn't `alphabetical` or size? I think you might want to either do a try/except or a manual `if order_by not in (RoomSortOrder.ALPHABETICAL.value, RoomSortOrder.SIZE.value):` or something